checkov/common/graph/graph_builder/variable_rendering/vertex_reference.py

```diff
@@ -1,11 +1,17 @@
-from typing import Union, List, Any
-from checkov.common.graph.graph_builder.graph_components.block_types import BlockType
+from __future__ import annotations
+
 from abc import ABC, abstractmethod
+from typing import Any, TypeVar, Generic, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from checkov.common.graph.graph_builder.graph_components.block_types import BlockType  # noqa
+
+_BlockT = TypeVar("_BlockT", bound="BlockType")
```

Comment:
Why do we need it?

Reply:
also here `VertexReference` can handle different types of `BlockType`, if it is in a context of Terraform it is a Terraform one and for CloudFormation a CloudFormation one. The `bound` limits the possibility to only allow types which subclass `common.BlockType`. This makes type resolution better down the line, because you know, which flavour exactly it is and not just it is a `common.BlockType` or a subclass of it.